packages/phishing-controller/src/PhishingDetector.test.ts

```diff
@@ -918,13 +939,85 @@ describe('PhishingDetector', () => {
+            );
 
             expect(result).toBe(true);
             expect(type).toBe('blocklist');
           },
         );
       });
+
+      it('blocks ipfs cid across various formats (cids located in subdomains and paths)', async () => {
+        // CID should not blocked
+        await withPhishingDetector(
+          [
+            {
+              allowlist: [],
+              blocklist: [
+                'QmUDBVyGwqKdSayk7kDKUaj9J41Ft1DWizcKUx5UmgMgGy',
+                'bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m',
+                'example.com',
+              ],
+              fuzzylist: [],
+              name: 'first',
+              tolerance: 2,
+              version: 1,
+            },
+          ],
+          async ({ detector }) => {
+            const { result, type } = detector.check(
+              formatHostnameToUrl(
+                'cf-ipfs.com/ipfs/bafybeiaysi4s6lnjev27ln5icwm6tueaw2vdykrtjkwiphwekaywqhcjze',
+              ),
+            );
+
+            expect(result).toBe(false);
+            expect(type).toBe('all');
+          },
+        );
+
+        // Gateways differ on where the CID is... sometimes in the path, sometimes in a magic subdomain
+        const expectedToBeBlocked = [
+          'ipfs.io/ipfs/bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m#x-ipfs-companion-no-redirect',
+          'gateway.pinata.cloud/ipfs/bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m#x-ipfs-companion-no-redirect',
+          'cloudflare-ipfs.com/ipfs/bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m#x-ipfs-companion-no-redirect',
+          'ipfs.eth.aragon.network/ipfs/bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m#x-ipfs-companion-no-redirect',
+          'bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m.ipfs.dweb.link/#x-ipfs-companion-no-redirect',
+          'bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m.ipfs.cf-ipfs.com/#x-ipfs-companion-no-redirect',
+          'example.com',
+          'example.com/foo/bar',
+        ];
+
+        // CID should be blocked
+        for await (const entry of expectedToBeBlocked) {
+          await withPhishingDetector(
+            [
+              {
+                allowlist: [],
+                blocklist: [
+                  'QmUDBVyGwqKdSayk7kDKUaj9J41Ft1DWizcKUx5UmgMgGy',
+                  'bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m',
+                  'example.com',
+                ],
+                fuzzylist: [],
+                name: 'first',
+                tolerance: 2,
+                version: 1,
+              },
+            ],
+            async ({ detector }) => {
+              const { result, type } = detector.check(
+                formatHostnameToUrl(entry),
+              );
+
+              expect(result).toBe(true);
+              expect(type).toBe('blocklist');
+            },
+          );
+        }
+      });
```

Comment:
Continuing from the above comment, this test would work to eliminate the final coverage problem and reach 100%:

``` suggestion
      });

      it('returns a result without a version when a config lacks a version and the blocklist contains an ipfs cid', async () => {
        await withPhishingDetector(
          [
            {
              allowlist: [],
              blocklist: [
                'bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m',
              ],
              fuzzylist: [],
              name: 'first',
              tolerance: 2,
            },
          ],
          async ({ detector }) => {
            const { result, version } = detector.check(
              formatHostnameToUrl(
                'ipfs.io/ipfs/bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m#x-ipfs-companion-no-redirect',
              ),
            );

            expect(result).toBe(true);
            expect(version).toBeUndefined();
          },
        );
      });